### PR TITLE
Update kubectl debug docs for 1.20 release

### PR DESCRIPTION
**Pull request by verb**
Documentation for kubernetes/features#1441 `kubectl debug` updates in 1.20.

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
lgtm from tech pov

**Comment by kbhawkey** [comment]
Hi, @verb .
There is also one reference to `kubectl alpha debug` in the concepts section,
`concepts\workloads\pods\ephemeral-containers.md`.

**Comment by verb** [comment]
This is ready for review. @soltysh, FYI I added much more content about debugging by copy and node debugging scenarios.

@yujuhong There's no a kubectl command to debug nodes by running a pod in the host namespaces. Would be happy to hear your opinion (or if we missed any problems with this approach) and whether you'd like me to update any of the node docs.

/hold cancel

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (kbhawkey) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:215]: nit: Line 210, extra space before `if`
nit: The `--share-processes` flag
nit: ... causes the Pod to use one process per namespace. You can identify other processes from other
containers using `ps`.

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (kbhawkey) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:271]: Line 271, Could you add a carriage return before, "You can prevent ..."

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (kbhawkey) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:293]: nit: Either use Ubuntu or add code formatting for the value as `ubuntu`.

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (kbhawkey) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:301]: nit: missing a back tick?

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (kbhawkey) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:329]: nit: s/&/and/

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (verb) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md]: Fixed the first two, but the third suggestion is incorrect. I've updated the text to:

> The `--share-processes` flag causes the Pod to use a single process namespace. You can identify other processes from other containers using `ps`.

I'm not sure if this addresses your initial concern, though.

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (verb) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md]: Wow. Yes. Not sure how that happened.

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (verb) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md]: Fixed.

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (verb) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md]: Added, thanks.

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (verb) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md]: Fixed.

**Comment by verb** [comment]
Hi @kbhawkey, thanks for the review. I've made the changes you requested.

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (kbhawkey) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md]: OK. I am not sure what exactly is a "single process namespace"?
Single process + one namespace or ?

Comment (verb) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md]: Under the hood it sets the `shareProcessNamespace` option in [v1.PodSpec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#podspec-v1-core). A "process namespace" is how the kernel keeps application processes in containers separate from processes in other containers and the larger system.

By default every container gets a separate "process namespace" so that there's no visibility to processes that are running in different containers. That makes debugging across container boundaries more difficult.

When one uses `--share-processes`, the kubelet will use one "process namespace" for all of the containers in the Pod so that processes from all of the containers in that Pod are visible to each other.

Comment (kbhawkey) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md]: Great. Thanks for the explanation!

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-24847--kubernetes-io-vnext-staging.netlify.app/docs/tasks/debug-application-cluster/debug-running-pod/

**Comment by annajung** [comment]
Hi @soltysh a lot of changes since your first review, can you please review again?
@yujuhong, also looking for your tech review, please see https://github.com/kubernetes/website/pull/24847#issuecomment-722369825

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (zacharysarah) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:266]: Make this a note.

```suggestion
{{< note >}}
```

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (zacharysarah) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:268]:
```suggestion
* To change the command of a specific container you must
```

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (zacharysarah) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:273]: This section is well constructed. Nice work! ✨

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (zacharysarah) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:274]:
```suggestion

{{< /note >}}

```

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (zacharysarah) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:312]: Grammar nit
```suggestion
If none of these approaches work, you can find the Node on which the Pod is
```

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (zacharysarah) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:328]: Avoid future tense
```suggestion
* `kubectl debug` automatically generates the name of the new Pod based on
```

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (zacharysarah) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:330]: Avoid future tense
```suggestion
* The container runs in the host IPC, Network, and PID namespaces.
```

**Review by zacharysarah — CHANGES_REQUESTED** [review]
@verb Good work here! ✨ Some small fixes required, otherwise LGTM from a docs perspective.

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (verb) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:275]: Thanks! I'm slowly getting a feel for website style 😀

**Review by verb — COMMENTED** [review]
Thanks for the review and suggestions, @zacharysarah!

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (verb) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md]: @kbhawkey Do you think I should rephrase/expand on this concept in the text or is it good as is?

**Comment by soltysh** [comment]
:+1: this still lgtm

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (kbhawkey) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md]: I'll read through the page ...

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (kbhawkey) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:176]: nit: [what situations?]: Sometimes Pod configuration options make it difficult to troubleshoot containers and applications running in containers?
nit: For example, you can't run `kubectl exec` to troubleshoot your container if your container image
does not include a shell or if your application crashes on startup.

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (kbhawkey) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:187]: nit: format `busybox`?

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (kbhawkey) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:209]: nit: You can remove the line before and after the note text.

Comment (kbhawkey) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:209]: nit:  You may want to add a single sentence to introduce the list of flag descriptions.

It may be helpful to provide more context about the `share-processes` flag.
What do you think about:


The `--share-processes` flag causes the Pod to share a single process namespace between all containers in a Pod.
A single process namespace lets a container view and signal processes from other containers in the same Pod.
You can identify processes from other containers using `ps`.
For more information about `shareProcessNamespace`, see the [PodSpec](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#podspec-v1-core) API.

From the API reference:
```
Share a single process namespace between all of the containers in a pod. When this is set containers will be able to view and signal processes from other containers in the same pod,
```

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (verb) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:176]: Applied the second suggestion, but I don't understand the first one. Is that the syntax for a reference link?

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (verb) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:187]: Fixed

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (verb) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:209]: * Removed the extra space
* The single sentence to introduce the list of flag descriptions was what another reviewer suggested I replace with the `note` tags. 😃  I'll leave this for now.
* For `--share-processes` I've replaced the description of the flag with a short reason why you'd want to use it here, and then linked to the `ShareProcessNamespace` docs for full context. I like the way it turned out, PTAL.

**Review by irvifa — COMMENTED** [review]
/approve

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (kbhawkey) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md]: Sorry @verb, I did not get a chance to look at the changes again before it was approved 😞

Comment (irvifa) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md]: Eh sorry... 😭

File: content/en/docs/tasks/debug-application-cluster/debug-running-pod.md
Comment (shubhamrajvanshi) [inline content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:232]: There is a typo here, it should **exit's** rather than **exists**

Comment (verb) [reply content/en/docs/tasks/debug-application-cluster/debug-running-pod.md:232]: @shubhamrajvanshi Thanks for pointing this out. I opened #25458 to fix.

**Pull request merged**
